From 23831177a4308fb18951537a0b3948e5547da02d Mon Sep 17 00:00:00 2001
From: Scott Tsai <scottt.tw@gmail.com>
Date: Sat, 22 Mar 2025 06:17:51 +0800
Subject: [PATCH] UtilsAvx512.cpp: fix r may be used uninitialized for gcc14

This version of fbgemm uses `-Werror` thus the warning breaks the build.
This is also reported upstream by others as https://github.com/pytorch/pytorch/issues/129358

ROCm/TheRock is building pytorch in Ubuntu 22.04 but Fedora 41 has a
newer gcc and would run into this bug.
---
 src/UtilsAvx512.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/UtilsAvx512.cc b/src/UtilsAvx512.cc
index cf00613d..c65eab87 100644
--- a/src/UtilsAvx512.cc
+++ b/src/UtilsAvx512.cc
@@ -920,6 +920,9 @@ static inline void transpose_contiguous_16x2_block(
     int64_t ld_dst,
     int mrem = 16) {
   __m512i r[2], d[2];
+  // Zero out r[] to avoid `may be used uninitialized` compilation error
+  r[0] = _mm512_setzero_si512();
+  r[1] = _mm512_setzero_si512();
   int i = 0;
   for (; (i + 1) * 16 <= mrem * 2; i++) {
     // normal load
-- 
2.49.0

